lib/events.js

```diff
@@ -93,121 +91,131 @@ MapboxEventManager.prototype = {
-     * @param {Object} err an error instance to log
-     * @private
-     */
-  handleError: function(err, callback){
+   * Handle an error that occurred while making a request
+   * @param {Object} err an error instance to log
+   * @private
+   */
+  handleError: function (err, callback) {
     if (callback) return callback(err);
   },
 
   /**
-     * Generate a session ID to be returned with all of the searches made by this geocoder instance
-     * ID is random and cannot be tracked across sessions
-     * @private
-     */
-  generateSessionID: function(){
-    var sha = crypto.createHash('sha256');
-    sha.update(Math.random().toString());
-    return sha.digest('hex')
+   * Generate a session ID to be returned with all of the searches made by this geocoder instance
+   * ID is random and cannot be tracked across sessions
+   * @private
+   */
+  generateSessionID: function () {
+    return new shajs.sha256().update(Math.random().toString()).digest('hex')
   },
 
   /**
-     * Get a user agent string to send with the request to the events service
-     * @private
-     */
-  getUserAgent: function(){
+   * Get a user agent string to send with the request to the events service
+   * @private
+   */
+  getUserAgent: function () {
     return 'mapbox-gl-geocoder.' + this.version + "." + navigator.userAgent;
   },
 
   /**
-     * Get the 0-based numeric index of the item that the user selected out of the list of options
-     * @private
-     * @param {Object} selected the geojson feature selected by the user
-     * @param {Object} geocoder a Mapbox-GL-Geocoder instance
-     * @returns {Number} the index of the selected result
-     */
-  getSelectedIndex: function(selected, geocoder){
+   * Get the 0-based numeric index of the item that the user selected out of the list of options
+   * @private
+   * @param {Object} selected the geojson feature selected by the user
+   * @param {Object} geocoder a Mapbox-GL-Geocoder instance
+   * @returns {Number} the index of the selected result
+   */
+  getSelectedIndex: function (selected, geocoder) {
     var results = geocoder._typeahead.data;
     var selectedID = selected.id;
-    var resultIDs = results.map(function (feature){
+    var resultIDs = results.map(function (feature) {
       return feature.id;
     });
     var selectedIdx = resultIDs.indexOf(selectedID);
     return selectedIdx;
   },
 
   /**
-     * Check whether events should be logged
-     * Clients using a localGeocoder or an origin other than mapbox should not have events logged
-     * @private
-     */
-  shouldEnableLogging: function(options){
-    if (!this.origin.includes('api.mapbox.com')) return false;
+   * Check whether events should be logged
+   * Clients using a localGeocoder or an origin other than mapbox should not have events logged
+   * @private
+   */
+  shouldEnableLogging: function (options) {
+    if (this.origin.indexOf('api.mapbox.com')  == -1) return false;
```

Comment:
Would `===` bet safer? It's what's used at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf#Checking_occurrences